pkg/controller/istiocsr/utils.go

```
@@ -165,164 +175,130 @@ func decodeDeploymentObjBytes(objBytes []byte) *appsv1.Deployment {
 		panic(fmt.Sprintf("unsupported object type: %T", desired))
 	}
-	return objectModified || objectMetadataModified(desired, fetched)
+}
+
+func checkRBACObjectSpecModified(desired, fetched client.Object) *bool {
+	switch d := desired.(type) {
+	case *rbacv1.ClusterRole:
+		result := checkRBACClusterRoleModified(d, fetched)
+		return &result
+	case *rbacv1.ClusterRoleBinding:
+		result := checkClusterRoleBindingModified(d, fetched.(*rbacv1.ClusterRoleBinding))
```

golangci-lint (2.5.0)

[error] right hand must be only type assertion

(forcetypeassert)